elements/pfe-icon/pfe-icon.ts

```diff
@@ -1,224 +1,157 @@
+  public static addIconSet(setName: string, getter: typeof PfeIcon['getIconUrl']) {
+    if (typeof getter !== 'function') {
+      Logger.warn('[PfeIcon.addIconSet(setName, getter)]: getter must be a function');
+    } else {
+      this.getters.set(setName, getter);
+      for (const instance of this.instances) {
+        instance.load();
+      }
     }
+  }
 
+  public static getIconUrl: URLGetter = (set: string, icon: string) =>
+    new URL(`./icons/${set}/${icon}.js`, import.meta.url);
 
-    // Register the icon set and set up the event indicating the change
-    this._iconSets.set(name, new PfeIconSet(name, path, resolveFunction));
+  private static onIntersect: IntersectionObserverCallback = records =>
+    records.forEach(({ isIntersecting, target }) => {
+      const icon = target as PfeIcon;
+      icon.#intersecting = isIntersecting;
+      ric(() => {
+        if (icon.#intersecting) {
+          icon.load();
+        }
+      });
+    });
 
-    document.body.dispatchEvent(
-      new CustomEvent('pfe-icon:add-icon-set', {
-        bubbles: false,
-        detail: {
-          set: this._iconSets.get(name),
-        },
-      })
-    );
-  }
+  private static io = new IntersectionObserver(PfeIcon.onIntersect);
 
-  private static _iconSets = new Map<string, PfeIconSet>();
+  private static getters = new Map<string, URLGetter>();
 
-  /**
-   * For example, `rh-leaf` loads a leaf icon from an icon set named "rh".
-   */
-  @observed
-  @property({ type: String, reflect: true }) icon = '';
-
-  /**
-   * The default size is 1em, so icon size matches text size. `2x`, etc, are multiples of font size. `sm`, `md`, etc are fixed pixel-based sizes.
-   */
-  @property({ type: String, reflect: true }) size: 'xl'|'lg'|'md'|'sm'|'1x'|'2x'|'3x'|'4x' = 'xl';
+  private static instances = new Set();
 
-  /**
-   * The color variant to use. This draws from your theming layer to color the icon.
-   * This will set icon color or background color (if circled is true).
-   * Values:
-   * - base
-   * - lightest
-   * - lighter
-   * - darker
-   * - darkest
-   * - complement
-   * - accent
-   * - critical
-   * - important
-   * - moderate
-   * - success
-   * - info
-   * - default
-   */
-  @property({ type: String, reflect: true }) color?: IconColor;
+  /** Icon set */
+  @property() set = PfeIcon.defaultIconSet;
 
-  @property({ type: String, reflect: true, attribute: 'on-fail' }) onFail?: 'collapse';
+  /** Icon name */
+  @observed
+  @property({ reflect: true }) icon = '';
 
-  /**
-   * Whether to draw a circular background behind the icon.
-   */
-  @property({ type: Boolean, reflect: true }) circled = false;
+  /** Size of the icon */
+  @property({ reflect: true }) size: 'sm'|'md'|'lg'|'xl' = 'sm';
 
-  @property({ type: Boolean, reflect: true }) block = false;
+  /** Accessible label for the icon */
+  @property({ reflect: true }) label?: string;
```

Comment:
remove, always aria-hidden=true the shadow icon, document to use `aria-label(ledby)`

cc Greg Gibson

Reply:
cc @hellogreg